feat: Adds max concurrent pull queries to limit effects of table scans (#7188)
confluentinc/ksql: docs/reference/server-configuration.md

interpreter gives significant performance gains. This can be disabled per query if the code compiler
is preferred.

## `ksql.query.pull.max.qps`

> Sets a query per second limit for pull queries. This limit is enforced per host (not per cluster).

Reviewer: ```suggestion
Sets a rate limit for pull queries, in queries per second. This limit is enforced per host, not per cluster.
```

> After hitting the limit, the host will fail pull query requests until it determines it's no longer

Reviewer: ```suggestion
After hitting the limit, the host will fail pull query requests until it determines that it's no longer
```

at the limit.

## `ksql.query.pull.max.concurrent.requests`

> Sets a concurrent query limit for pull queries. This limit is enforced per host (not per cluster).

Reviewer: ```suggestion
Sets the maximum number of concurrent pull queries. This limit is enforced per host, not per cluster.
```

> After hitting the limit, the host will fail pull query requests until it determines it's no longer

Reviewer: ```suggestion
After hitting the limit, the host will fail pull query requests until it determines that it's no longer
```
